File: rules/non-determinism/java/use-current-time.java
public void Test()
{
    var sut = new CustomerService();

    Customer customer = sut.Create("Name");

    // rule-id: use-current-time
    Assert.Equal(DateTime.Now, customer.DateCreated);
}

public void Test()
{
    DateTime now = DateTime.Now;
    var sut = new CustomerService();

    Customer customer = sut.Create("Name", now);

    // rule-id: use-current-time
    Assert.Equal(now, customer.DateCreated);
}

File: rules/non-determinism/python/use-current-time.py
import random
import time
from datetime import datetime

t = datetime.now()
# rule-id: use-current-time
assert(t > 0)
# rule-id: use-current-time
assert(t < 0)
# rule-id: use-current-time
assert(t == 0)
# rule-id: use-current-time
assert(t >= 0)
# rule-id: use-current-time
assert(t <= 0)
# rule-id: use-current-time
assert(t <> 0)
# rule-id: use-current-time
assert(t != 0)

t = time.now()
# rule-id: use-current-time
assert(t > 0)
# rule-id: use-current-time
assert(t < 0)
# rule-id: use-current-time
assert(t == 0)
# rule-id: use-current-time
assert(t >= 0)
# rule-id: use-current-time
assert(t <= 0)
# rule-id: use-current-time
assert(t <> 0)
# rule-id: use-current-time
assert(t != 0)
